src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/INodeWithDebugInfo.cs

```diff
@@ -15,13 +15,15 @@ public struct DebugLocInfo
         public readonly string FileName;
         public readonly int LineNumber;
         public readonly int ColNumber;
+        public readonly NativeLocInfo OffsetMapping;
 
-        public DebugLocInfo(int nativeOffset, string fileName, int lineNumber, int colNumber = 0)
+        public DebugLocInfo(int nativeOffset, string fileName, NativeLocInfo offsetMapping, int lineNumber, int colNumber = 0)
         {
             NativeOffset = nativeOffset;
             FileName = fileName;
             LineNumber = lineNumber;
             ColNumber = colNumber;
+            OffsetMapping = offsetMapping;
```

Comment:
Isn't the OffsetMapping thing redundant with NativeOffset and LineNumber?

If you feel like you need to do unnatural things, I would just stop using the `INodeWithDebugInfo` with ReadyToRun backend - the way we publish from CorInfoImpl for ReadyToRun can be completely different. We already use a different class to represent the method code and EH info anyway. These data structures are geared towards the debugging info we emit for full AOT. It would be unfortunate to waste memory only because we want to share what is awkward to share.

Reply:
Yes that was awkward. I have split out the jit / r2r parts into their respective CorInfoImpl partial classes so we only carry the info we need for each.